tfjs-backend-webgpu/src/webgpu_util.ts

```diff
@@ -58,6 +58,30 @@ export function computeDispatch(
   ];
 }
 
+export function computeWorkGroupSizeForConv2d(
+    layout: {x: number[], y?: number[], z?: number[]},
+    outputShape: number[]): [number, number, number] {
+  const dim1 = arrayProduct(layout.y.map(d => outputShape[d]));
+  // TODO(jiajia.qin@intel.com): More fine tune based on outputShape.
+  if (dim1 <= 4) {
+    return [16, 4, 1];
+  }
+
+  return [16, 16, 1];
+}
+
+export function computeWorkPerThreadForConv2d(
+    layout: {x: number[], y?: number[], z?: number[]},
+    outputShape: number[]): [number, number, number] {
+  const dim1 = arrayProduct(layout.y.map(d => outputShape[d]));
+  // TODO(jiajia.qin@intel.com): More fine tune based on outputShape.
+  if (dim1 <= 4) {
```

Comment:
Don't we want to check that dim0 is significantly bigger than dim1? And why do we never want to return square workPerThread (where the first and second values of the returned array match)?

Reply:
The current rule is still not complete. I saw [2, 4, 1] was better than [2, 2, 1] for PoseNet demo on Intel GPU. So I just set it to [2, 4, 1]. Anyway, I add more rules in newest change to include the square workPerThread.
And I think we need to continue to make the rules better. But we can put it in later work to make this patch simple. How do you think?